src/Logging/DataQueue.cpp

```diff
@@ -61,6 +61,14 @@ size_t DataQueue::getDataSize() {
 	return _writer->dataSize();
 }
 
+size_t DataQueue::getNumEventsInQueue() {
+	return _publishQueue->getNumEvents();
+}
+
+bool DataQueue::isCacheFull() {
+	return _publishQueue->getNumEvents() == _publishQueue->getFileQueueSize() + RAM_QUEUE_EVENT_COUNT;
```

Comment:
Any chance at all that getNumEvents() could be greater than the right side of the equation? I always like to make the operator >= in cases like this. I can't trust another person to properly code a class, and I definitely cannot trust myself to properly code a class.